addons/barcodes_gs1_nomenclature/static/src/js/tests/epc_utils.test.js

```diff
@@ -0,0 +1,34 @@
+import { expect, test } from "@odoo/hoot";
+import { decode } from "@barcodes_gs1_nomenclature/js/epc_utils";
+
+test.tags("RFID");
+
+test("Decode SGTIN-96", () => {
+    const test = decode("3066C4409047E140075BCD15");
+    expect(test).toEqual("urn:epc:id:sgtin:95060001343.05.123456789");
+});
+
+test("Decode SGTIN-198", () => {
+    const test = decode("3666C4409047E159B2C2BF100000000000000000000000000000");
+    expect(test).toEqual("urn:epc:id:sgtin:95060001343.05.32a%2Fb");
+});
+
+test("Decode SSCC-96", () => {
+    const test = decode("311BA1B300CE0A6A83000000");
+    expect(test).toEqual("urn:epc:id:sscc:952012.03456789123");
+});
+
+test("Decode SGLN-96", () => {
+    const test = decode("3276451FD46072000000162E");
+    expect(test).toEqual("urn:epc:id:sgln:9521141.12345.5678");
+});
+
+test("Decode SGLN-195", () => {
+    const test = decode("3976451FD46072CD9615F8800000000000000000000000000000");
+    expect(test).toEqual("urn:epc:id:sgln:9521141.12345.32a%2Fb");
+});
+
+test("Decode non hex value", () => {
+    const test = decode("006604409047014007500015");
+    expect(test).toEqual("006604409047014007500015");
+});
```

Comment:
```suggestion
test("Decode unrecognized EPC", () => {
    const test = decode("006604409047014007500015");
    expect(test).toEqual("006604409047014007500015");
});

test("Decode non hex value", () => {
    const test = decode("300123456789ABCDEFGHIJ");
    expect(test).toEqual("300123456789ABCDEFGHIJ");
});
```
En vrai 006604409047014007500015 est un nombre hexadécimal valide, ce test c'est plutôt : "Decode unrecognized EPC" car 66 (les 0 sont perdus à la conversion en BigInt)  n'est pas reconnu comme un header valide.
Ce test couvre la condition suivante : 
```javascript
    //Find template
    if (header in SCHEME_TEMPLATE && SCHEME_TEMPLATE[header] !== undefined) {
        var template = SCHEME_TEMPLATE[header];
    } else {
        return hexString; // return original value if unable to decode
    }
```

Par contre on peut ajouter le nouveau test pour couvrir le try/catch du parsing en BigInt :
```javascript
    // Prepare values
    var hexValue;
    try{
        hexValue = BigInt("0x" + hexString);
    }catch(e){
        return hexString; // return original value if unable to decode
    }
```